src/GitTfs/Core/GitTfsRemote.cs

```diff
@@ -875,8 +875,11 @@ private string BuildCommitMessage(string tfsCheckinComment, int changesetId)
         {
             var builder = new StringWriter();
             builder.WriteLine(tfsCheckinComment);
-            builder.WriteLine(GitTfsConstants.TfsCommitInfoFormat,
-                TfsUrl, TfsRepositoryPath, changesetId);
+            builder.WriteLine(GitTfsConstants.TfsCommitInfoFormat, TfsUrl, TfsRepositoryPath, changesetId);
+            var match = Regex.Match(TfsRepositoryPath, @"\$/(?<TeamProject>[^/]*?)/?");
+            if (match.Success)
+                builder.WriteLine(GitTfsConstants.TfsCommitUrlFormat, TfsUrl, match.Groups["TeamProject"].Value, changesetId);
+
```

Comment:
Haven't tried it yet, but from the description it will give you the full url of the changeset, so one can use the url to open it in a browser. Personally, I would like this feature.
On the other hand, what worries me a little about this change is that this will cause the git object IDs (SHA1) to change for every commit, depending on which version of git-tfs was used to import the change.

 I see these options here:
- Don't allow adding this feature
- Accept that we have changed object IDs with different git-tfs versions. This would make exchanging commits offline more difficult, as now all involved parties need to have the same git tfs version
- Use https://git-scm.com/docs/git-notes to do this and store that information in a note. Notes can selectively be shown or not when showing a commit. We are already doing this for other information like review information already.
- Do this completely outside of git-tfs, e.g. in https://git-scm.com/docs/git-filter-branch or its succesor https://github.com/newren/git-filter-repo by filtering the repo after the fact. As far as I can see, all the information is already available in the commit message to create the full URL

Reply:
Hmmm, I did not think about the SHA changing because of that change. The url can easily be constructed from the information that was there. You just have to know the format.  I am fine with any of your secondary options.

Reply:
For me this information could be added to the commit message when we use the `--export` option that is explicitly there to export this kind of metadata. 
https://github.com/git-tfs/git-tfs/blob/master/doc/commands/clone.md#export-metadatas
And I also think that these information are stored using git-notes if the `--export` option is not used (but I'm not really sure. We should look at the code).

So it could make perfectly sense to add this information there.

And we could accept that the sha1 change when using this option. We should just ensure that the sha1 don't change in the normal case when we don't use this option.

Reply:
Yes, your suggestion is what I meant with my 3rd point in the comment above, cited for reference here:

> * Use https://git-scm.com/docs/git-notes to do this and store that information in a note. Notes can selectively be shown or not when showing a commit. We are already doing this for other information like review information already.

And you are also correct that we already use git-notes for storing this metadata, so there is already existing code dealing with git notes.